sort/cocktailsort.go

```diff
@@ -0,0 +1,41 @@
+// Implementation of Cocktail sorting
+// reference: https://en.wikipedia.org/wiki/Cocktail_shaker_sort
+
+package sort
+
+import "github.com/TheAlgorithms/Go/constraints"
+
+// Cocktail sort is a variation of bubble sort, operating in two directions (beginning to end, end to beginning)
+func Cocktail[T constraints.Integer](arr []T) []T {
+	if len(arr) == 0 { // ignore 0 length arrays
+		return arr
+	}
+
+	swapped := false // true if swapped two or more elements in the last loop
+	// if it loops through the array without swapping, the array is sorted
+
+	for ok := true; ok; ok = swapped { // exits when swapped == false
```

Comment:
Isn't this whole mess just equivalent to:
```go
for swapped {
```
Not sure what I am missing.

I guess that's to ensure the loop doesn't break in the first iteration? That can be fixed by simply setting the initial value of `swapped` to `true`.